### PR TITLE
Implement Anthropic prompt caching

### PR DESCRIPTION
**Pull request by rowillia**
Release Notes:

- Adds support for Prompt Caching in Anthropic.  For models that support it this can dramatically lower cost while improving performance.

### CODE REVIEW

**Review by rgbkrk — APPROVED** [review]
Rad!

File: crates/language_model/src/request.rs [hunk 1]
```diff
@@ -296,18 +297,50 @@ impl LanguageModelRequest {
...
+                        Some(anthropic::CacheControl {
+                            cache_type: anthropic::CacheControlType::Ephemeral,
+                        })
+                    } else {
+                        None
+                    };
+                    let anthropic_message_content: Vec<anthropic::Content> = message
+                        .content
+                        .into_iter()
+                        // TODO: filter out the empty messages in the message construction step
```
Comment (rgbkrk) [inline crates/language_model/src/request.rs:310]: This conflicts now but I started this in https://github.com/zed-industries/zed/pull/16270. Happy to finish it up.

Comment (rowillia) [reply crates/language_model/src/request.rs:310]: @rgbkrk that'd be great if you don't mind - once this lands I was also gonna take a run at enabling caching for the inline assistant prompt as well.

Comment (rgbkrk) [reply crates/language_model/src/request.rs:310]: Cool, I'll merge this now and then rebase my branch against yours.

**Pull request merged**